File: app/controllers/main_controller.py
from flask import Blueprint, render_template, redirect, url_for

# Criar Blueprint para rotas principais
bp = Blueprint('main', __name__)


@bp.route('/')
def index():
    """Página inicial - redireciona para dashboard se logado"""
    from flask import session
    
    # Se usuário estiver logado, redirecionar para dashboard
    if session.get('user_id'):
        return redirect(url_for('dashboard.dashboard'))
    
    return render_template('main/index.html', title='Home')


@bp.route('/about')
def about():
    """Página sobre"""
    return render_template('main/about.html', title='Sobre')

File: src/main/java/com/torneseumprogramador/desafioiamonolito/controllers/MainController.java
package com.torneseumprogramador.desafioiamonolito.controllers;

import jakarta.servlet.http.HttpSession;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

/**
 * Controller para rotas principais
 */
@Controller
public class MainController {

    @GetMapping("/")
    public String index(HttpSession session, Model model) {
        // Se usuário estiver logado, redirecionar para dashboard
        if (session.getAttribute("user_id") != null) {
            return "redirect:/dashboard";
        }
        
        model.addAttribute("title", "Home");
        return "main/index";
    }

    @GetMapping("/about")
    public String about(Model model) {
        model.addAttribute("title", "Sobre");
        return "main/about";
    }
}
